flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/common/CommonExecSinkITCase.java

```diff
@@ -0,0 +1,315 @@
+                        Row.of(3, "foo", Instant.parse("2020-11-11T10:11:22.777Z")),
+                        Row.of(4, "foo", Instant.parse("2020-11-11T10:11:23.888Z")));
+
+        final TableDescriptor sourceDescriptor =
+                TableFactoryHarness.newBuilder()
+                        .schema(schema)
+                        .source(
+                                new TimestampTestSource(
+                                        rows, ROW(INT(), STRING(), TIMESTAMP_LTZ(3))))
+                        .sink(
+                                new TableFactoryHarness.SinkBase() {
+                                    @Override
+                                    public DynamicTableSink.SinkRuntimeProvider
+                                            getSinkRuntimeProvider(
+                                                    DynamicTableSink.Context context) {
+                                        return SinkProvider.of(
+                                                TestSink.newBuilder()
+                                                        .setWriter(new TestWriter(timestamps))
+                                                        .setCommittableSerializer(
+                                                                TestSink.StringCommittableSerializer
+                                                                        .INSTANCE)
+                                                        .build());
+                                    }
+                                })
+                        .build();
+        tableEnv.createTable("T1", sourceDescriptor);
+        String sqlStmt = "INSERT INTO T1 SELECT * FROM T1";
+        assertPlan(tableEnv, sqlStmt, true);
+        tableEnv.executeSql(sqlStmt).await();
+        assertTimestampResults(timestamps, rows);
+    }
+
+    @Test
+    public void testStreamRecordTimestampInserterDataStreamSinkProvider()
+            throws ExecutionException, InterruptedException {
+        final StreamTableEnvironment tableEnv = StreamTableEnvironment.create(env);
+        final SharedReference<List<Long>> timestamps = sharedObjects.add(new ArrayList<>());
+        final Schema schema = schemaForTimeOperatorTesting(true);
+        final List<Row> rows =
+                Arrays.asList(
+                        Row.of(1, "foo", Instant.parse("2020-11-10T11:34:56.123Z")),
+                        Row.of(2, "foo", Instant.parse("2020-11-10T12:34:56.789Z")),
+                        Row.of(3, "foo", Instant.parse("2020-11-11T10:11:22.777Z")),
+                        Row.of(4, "foo", Instant.parse("2020-11-11T10:11:23.888Z")));
+
+        final TableDescriptor sourceDescriptor =
+                TableFactoryHarness.newBuilder()
+                        .schema(schema)
+                        .source(
+                                new TimestampTestSource(
+                                        rows, ROW(INT(), STRING(), TIMESTAMP_LTZ(3))))
```

Comment:
use `schema.toPhysicalRowDataType`? this would also avoid the mismatch that currently exists in `TIMESTAMP` vs `TIMESTAMP_LTZ`

Reply:
I'm a bit confused, `schema` is a `Schema` instance, and the `toPhysicalRowDataType` is on `ResolvedSchema`.

Reply:
Changed to create a ResolvedSchema and a Schema from it.